src/textual/widgets/_month_calendar.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import annotations
+
+import calendar
+import datetime
+
+from rich.text import Text
+from textual.app import ComposeResult
+from textual.reactive import Reactive, var
+from textual.widget import Widget
+from textual.widgets import DataTable
+
+
+class InvalidMonthNumber(Exception):
+    pass
+
+
+class InvalidWeekdayNumber(Exception):
+    pass
+
+
+class MonthCalendar(Widget):
+    year: Reactive[int | None] = Reactive[int | None](None)
+    month: Reactive[int | None] = Reactive[int | None](None)
+    first_weekday: Reactive[int] = Reactive(0)
+    _calendar: var[calendar.Calendar] = var(calendar.Calendar())
+
+    def __init__(
+        self,
+        year: int | None = None,
+        month: int | None = None,
+        first_weekday: int = 0,
+        *,
+        name: str | None = None,
+        id: str | None = None,
+        classes: str | None = None,
+        disabled: bool = False,
+    ) -> None:
+        super().__init__(name=name, id=id, classes=classes, disabled=disabled)
+        self.year = year
+        self.month = month
+        self.first_weekday = first_weekday
+
+    def compose(self) -> ComposeResult:
+        yield DataTable()
```

Comment:
Open thoughts: I wonder if we should display the month name above the DataTable. I wonder if it should be something that can be toggled.

Reply:
I'm assuming this would simply display "November 2023" for example, and then if developers wanted something different  they would be expected to create a custom widget and set `MonthCalendar.month_header` to False?

Reply:
I'm not sure - I guess the month name would be defined by locale too? @willmcgugan any thoughts?